src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -699,14 +683,14 @@ class Peer(
         Secp256k1DHFunctions, Chacha20Poly1305CipherFunctions, SHA256HashFunctions
     )
 
-    private suspend fun run() {
+    private suspend fun run(swapInManager: SwapInManager) {
         logger.info { "peer is active" }
         for (event in input) {
-            processEvent(event)
+            processEvent(swapInManager, event)
         }
     }
 
-    private suspend fun processEvent(cmd: PeerCommand) {
+    private suspend fun processEvent(swapInManager: SwapInManager, cmd: PeerCommand) {
```

Comment:
> I get that we need the initial channels though, but I think this could work:

Where do you hold the reference to the `SwapInManager` if you do that? Especially since it needs to be initialized with the list of channels restored from the DB, I don't see how we can make that work? This is the main issue with our model where many things that are actually interdependent are initialized asynchronously in different coroutines...

Reply:
> Where do you hold the reference to the SwapInManager if you do that?

In the `TrySwapIn`, sorry my snippet was incomplete.

Reply:
Interesting, that's a good idea, I'll look into it!

Reply:
It has a drawback though, it's not very future-proof because that means we can't let the wallet application send `TrySwapIn` to trigger a new attempt (for example after a liquidity policy update), since they won't be able to get a reference to that `SwapInManager`...

Reply:
I change the `SwapInCommand` to be processed in a separate coroutine (and not treated as a `PeerCommand`), which makes this cleaner.